git/working.go

```diff
@@ -169,22 +128,42 @@ func (c *Checkout) GetNote(ctx context.Context, rev string, note interface{}) (b
 	return getNote(ctx, c.dir, c.realNotesRef, rev, note)
 }
 
+// HeadRevision returns the revision of the current git HEAD
 func (c *Checkout) HeadRevision(ctx context.Context) (string, error) {
 	return refRevision(ctx, c.dir, "HEAD")
 }
 
-func (c *Checkout) SyncRevision(ctx context.Context) (string, error) {
-	return refRevision(ctx, c.dir, c.config.SyncTag)
+// GetSyncMarkerRevision returns the revision of the SyncMarker
+func (c *Checkout) GetSyncMarkerRevision(ctx context.Context) (string, error) {
+	if c.config.ReadOnly || c.config.StateMode == entities.NativeStateMode {
+		return GetNativeSyncMarkerRevision(ctx)
+	}
+	return getGitSyncTagRevision(ctx, c.dir, c.config.SyncTag)
 }
 
-func (c *Checkout) MoveSyncTagAndPush(ctx context.Context, tagAction TagAction) error {
-	if tagAction.SigningKey == "" {
-		tagAction.SigningKey = c.config.SigningKey
+// UpdateSyncMarker updates the state that Flux relies on for keeping track of the so-called "High Water Mark" for which commits Flux has reconciled.
+func (c *Checkout) UpdateSyncMarker(ctx context.Context, syncMarkerAction SyncMarkerAction) error {
+	if syncMarkerAction.SigningKey == "" {
+		syncMarkerAction.SigningKey = c.config.SigningKey
+	}
+	if c.config.ReadOnly || c.config.StateMode == entities.NativeStateMode {
+		return updateNativeSyncMarker(ctx, syncMarkerAction)
 	}
-	return moveTagAndPush(ctx, c.dir, c.config.SyncTag, c.upstream.URL, tagAction)
+	return updateGitSyncTag(ctx, c.dir, c.config.SyncTag, c.upstream.URL, syncMarkerAction)
 }
 
+// VerifySyncTag validates the gpg signature created by git tag.
 func (c *Checkout) VerifySyncTag(ctx context.Context) error {
+	// READONLY-NOTE: we're going to have to figure out how this makes sense in the context of a readonly repo.  There is a `git verify-commit` command to go with `git verify-tag`, but I'm guessing that in this case we are hoping/assuming that someone will never want to verify the tag in readonly mode since the fact that we're in readonly mode confirms the tag not to exist.  On the other hand, if there is another fluxd running that can push the tag perhaps a readonly flux instance may want to be able to check it.  In that case... I donno.
```

Comment:
ok.  I suspected so.  it was the case you had mentioned where there are multiple flux daemons running on the same repo, some readonly and one read/write.  I'll remove the note. thanks.

Reply:
if you would like me to also remove the sanity checks I can do that too.  To be clear, I'm talking about these lines:

```
	if c.config.ReadOnly {
		return fmt.Errorf("verifying the Git Sync Tag in readonly mode is not supported")
	}

	if c.config.StateMode == entities.NativeStateMode {
		return fmt.Errorf("verifying the Git Sync Tag is not supported when using --flux-state-mode=" + entities.NativeStateMode)
	}
```

Reply:
I removed the lines mentioned above to avoid further discussion on it (especially considering doing this extra level of checking goes well beyond the current standard I think is most appropriate for the git module to be aware of)